test/basic_get_save_v1_test.py

```diff
@@ -168,6 +195,128 @@ def getRhodobacterRef(self):
     def getType(self, ref=None):
         return self.ws.get_object_info_new({"objects": [{"ref": ref}]})[0][2]
 
+    def _downgraded(self, data):
+        self.assertTrue('features' in data)
+        self.assertTrue('cdss' not in data)
+        self.assertTrue('mrnas' not in data)
+        one_feat = data['features'][0]
+        self.assertEqual(one_feat['type'], 'gene')
+        self.assertEqual(one_feat['function'], 'leader; Amino acid biosynthesi'
+                                               's: Threonine; product:thr oper'
+                                               'on leader peptide')
+        self.assertEqual(one_feat['aliases'][0], 'ECK0001; JW4367')
+        self.assertEqual(one_feat['ontology_terms'],
+                         {'GO':
+                             {'GO:0009088':
+                                 {
+                                     "evidence": [],
+                                     "id": "GO:0009088",
+                                     "ontology_ref": "6308/3/2",
+                                     "term_lineage": [],
+                                     "term_name": "threonine biosynthetic process"
+                                 }}})
+        two_feat = data['features'][-1]
+        self.assertEqual(two_feat['type'], 'gene')
+        self.assertEqual(two_feat['aliases'][0], 'b4370')
+
+    @log
+    def test_genome_downgrade(self):
+        data = json.load(open('data/new_ecoli_genome.json'))
+        down_data = GenomeInterfaceV1.downgrade_genome(data)
+        self._downgraded(down_data)
+
+    def test_bad_get_genome_input(self):
+        with self.assertRaises(ValueError):
+            ret = self.impl.get_genome_v1(self.ctx,
+                      {
+                          'genomes': [{
+                              'ref': self.new_genome_ref
+                          }],
+                          'no_data': 'T'
+                      })[0]
+        with self.assertRaises(ValueError):
+            ret = self.impl.get_genome_v1(self.ctx,
+                      {
+                          'genomes': [{
+                              'ref': self.new_genome_ref
+                          }],
+                          'ignoreErrors': 'T'
+                      })[0]
+        with self.assertRaises(ValueError):
+            ret = self.impl.get_genome_v1(self.ctx,
+                      {
+                          'genomes': [{
+                              'ref': self.new_genome_ref
+                          }],
+                          'downgrade': 'T'
+                      })[0]
+        with self.assertRaises(ValueError):
+            ret = self.impl.get_genome_v1(self.ctx,
+                      {
+                          'genomes': [{
+                              'ref': self.new_genome_ref
+                          }],
+                          'no_metadata': 'T'
+                      })[0]
+
```

Comment:
Adding this 'downgrade' variable and set it default to true shouldn't change the behavior of get_genome_v1 on old genome type object. I think you should add one more test. For old genome type, get_genome_v1 should return same for 'downgrade' is 0 and 1.

Reply:
Added to "test_get_all"